File: Python/Samples/Observer/UtObserver.py
import unittest
from Observers import Observer, ObserverMailServer, ObserverPbx
from Subjects import Subject, SubjectEflow


class UtVisitor(unittest.TestCase):

    def test_observer(self):

        # Create observers
        pbx = ObserverPbx()
        ms = ObserverMailServer()
            
        # Create subject
        subject = SubjectEflow()
        subject.attach(pbx)
        subject.attach(ms)

        # Notify when JB is leave of absence
        subject.notify("JB", "Hachi")
            
        self.assertTrue(True)


if __name__ == '__main__':
    unittest.main()

File: CSharp/DP.UnitTest/UtObserver.cs
using System;
using System.Diagnostics;
using DP.UnitTest.Utility;
using Xunit;
using Xunit.Abstractions;
using DP.Domain.Samples.Observer;

namespace DP.UnitTest
{
    public class UtObserver
    {
        private readonly ITestOutputHelper _output;

        public UtObserver(ITestOutputHelper output)
        {
            this._output = output;
            System.Diagnostics.Trace.Listeners.Add(new XunitTraceListener(this._output)); 
        }

        [Fact] 
        public void TestObserver()
        {
            //Create observers
            IObserver pbx = new ObserverPbx();
            IObserver ms = new ObserverMailServer();
            
            //Create subject
            ISubject subject = new SubjectEflow();
            subject.Attach(pbx);
            subject.Attach(ms);

            //Notify when JB is leave of absence
            subject.Notify("JB", "Hachi");
            

            Assert.True(true);
        }
    }
}
